sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamValuesRel.java

```diff
@@ -58,28 +59,36 @@ public BeamValuesRel(
 
   }
 
-  @Override public PCollection<Row> buildBeamPipeline(
-      PCollectionTuple inputPCollections,
-      BeamSqlEnv sqlEnv) throws Exception {
+  @Override
+  public PTransform<PCollectionTuple, PCollection<Row>> toPTransform(BeamSqlEnv sqlEnv) {
```

Comment:
I think that `sqlEnv` is known at creation time of the `BeamRelNodes`, probably would make sense to make it less lazy and move it to `RelNode` constructor

Reply:
Ah, great. I will try that.

Reply:
OK - I added it in a second commit since it is maybe not perfect. I removed the parameter there. Some of the `BeamRelNode` implementations do use it so I plumbed it through the rule set. I think that makes some sense - probably also the conditionals in joins should be handled by rules and the way that the env is used in sources and sinks should work differently but I'm not sure how just yet.